include/laser_filters/intensity_filter.h

```diff
@@ -88,17 +88,21 @@ class LaserScanIntensityFilter : public filters::FilterBase<sensor_msgs::LaserSc
     {
       if (filtered_scan.intensities[i] <= lower_threshold_ ||                           // Is this reading below our lower threshold?
           filtered_scan.intensities[i] >= upper_threshold_)                             // Is this reading above our upper threshold?
-      {      
-
+      {
         filtered_scan.ranges[i] = std::numeric_limits<float>::quiet_NaN();              // If so, then make it an invalid value (NaN)
       }
 
-      if( isinf((double)filtered_scan.intensities[i]) || isnan((double)filtered_scan.intensities[i]) )
-	continue;
-      int cur_bucket = (int) ((std::min((double)filtered_scan.intensities[i], hist_max)/hist_max)*num_buckets) ;
-      if (cur_bucket >= num_buckets-1)
-	cur_bucket = num_buckets-1 ;
-      histogram[cur_bucket]++ ;
+
+      if (disp_hist_){
```

Comment:
Why are you only doing this computation when `disp_hist` is non-zero? From looking at the code, I think `disp_hist` is mean to be a flag about whether or not to print out the histogram. Shouldn't the histogram be updated regardless?